apps/aecore/src/aec_events.erl

```diff
@@ -16,9 +16,10 @@
 
 -include("blocks.hrl").
 
--type event() :: block_created
-               | start_mining
-               | top_changed
+-type event() :: start_mining
+               | block_created %% We created (mined) a new block. It usually becomes the new top block, though not necessarily.
+               | top_synced %% We received a block, pulled from a network peer, that became the new top.
+               | top_changed %% We received a block, pushed by a network peer, that became the new top.
```

Comment:
@uwiger @velzevur FYI In this WIP PR I documented the current semantics of `block_created` and `top_changed` internal events. I also added a new `top_synced` event for symptom of block candidate generation not being preempted when node pulling blocks from network (at bootstrap or node restart).

In the commit message I listed other modules that may benefit from this internal event, though so far I do not plan to change behaviour of those (e.g. on bootstrap lots of blocks get pulled from network so it would be a big difference).

Reply:
This would imply that `aesc_fsm_min_depth_watcher.erl` should also be extended to monitor `top_synced`?
https://github.com/aeternity/epoch/blob/master/apps/aechannel/src/aesc_fsm_min_depth_watcher.erl#L36

Reply:
Opened as separate PR #1210